src/lib.rs

```diff
@@ -105,6 +111,22 @@ impl<'a> TreeItem<'a> {
         }
     }
 
+    pub fn child(&self, index: usize) -> Option<&Self> {
+        if index < self.children.len() {
+            Some(&self.children[index])
+        } else {
+            None
+        }
+    }
+
+    pub fn child_mut(&mut self, index: usize) -> Option<&mut Self> {
+        if index < self.children.len() {
+            Some(&mut self.children[index])
+        } else {
+            None
+        }
```

Comment:
See other comment
```suggestion
        self.children.get_mut(index)
```

Reply:
This was a small oversight on my part. Using `.get_mut()` indeed makes the intent clearer.